network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TerminalRefUtils.java

```diff
@@ -18,18 +19,30 @@ private TerminalRefUtils() {
     }
 
     public static Terminal getTerminal(NetworkObjectIndex index, TerminalRefAttributes terminalRefAttributes) {
+        return getTerminal(index, terminalRefAttributes, null);
+    }
+
+    public static Terminal getTerminal(NetworkObjectIndex index, TerminalRefAttributes terminalRefAttributes, ResourceType resourceType) {
         if (terminalRefAttributes == null) {
             return null;
         }
-        Identifiable<?> identifiable = index.getIdentifiable(terminalRefAttributes.getConnectableId());
+        Identifiable<?> identifiable;
+        if (resourceType == null) {
```

Comment:
We can add the resource type and getIdentifiable optimisation in a separate PR (maybe just implement preloading collection for getIdentifiable)